File: CS/CustomWizardExample/Wizard/Presenters/ChoosePersonsPage.cs
using DevExpress.Data.WizardFramework;
using DevExpress.DataAccess.Wizard.Model;
using CustomWizardExample.Wizard.Views;
using System.Windows.Forms;
using DevExpress.Snap.Services;

namespace CustomWizardExample.Wizard.Presenters {
    public class ChoosePersonsPage : WizardPageBase<IChoosePersonsPageView, DataSourceModel> {
        public ChoosePersonsPage(IChoosePersonsPageView view) : base(view) { }

        public override void Begin() {
            View.PersonSourceType = PersonSourceType.XmlFilePath;
        }

        public override bool Validate(out string errorMessage) {
            errorMessage = string.Empty;
            bool result = false;
            if (View.PersonSourceType == PersonSourceType.Static) {
                result = View.ShowWarning();
                if (result)
                    View.XmlFilePath = "Data.xml";
            }
            else {
                using (OpenFileDialog dlg = new OpenFileDialog()) {
                    dlg.Filter = "XML Files|*.xml";
                    if (dlg.ShowDialog() == DialogResult.OK) {
                        result = true;
                        View.XmlFilePath = dlg.FileName;
                    }
                }
            }
            return result;
        }

        public override void Commit() {
            SnapDataSourceModel snapDataSourceModel = (SnapDataSourceModel)Model;
            snapDataSourceModel.Tag = View.XmlFilePath;
            snapDataSourceModel.DataSchema = new Person();
        }
        public override bool MoveNextEnabled { get { return false; } }
        public override bool FinishEnabled { get { return true; } }
    }
}

File: VB/CustomWizardExample/Wizard/Presenters/ChoosePersonsPage.vb
Imports DevExpress.Data.WizardFramework
Imports DevExpress.DataAccess.Wizard.Model
Imports CustomWizardExample.Wizard.Views
Imports System.Windows.Forms

Namespace CustomWizardExample.Wizard.Presenters
    Public Class ChoosePersonsPage
        Inherits WizardPageBase(Of IChoosePersonsPageView, DataSourceModel)

        Public Sub New(ByVal view As IChoosePersonsPageView)
            MyBase.New(view)
        End Sub

        Public Overrides Sub Begin()
            View.PersonSourceType = PersonSourceType.XmlFilePath
        End Sub

        Public Overrides Function Validate(<System.Runtime.InteropServices.Out()> ByRef errorMessage As String) As Boolean
            errorMessage = String.Empty
            Dim result As Boolean = False
            If View.PersonSourceType = PersonSourceType.Static Then
                result = View.ShowWarning()
                If result Then
                    View.XmlFilePath = "Data.xml"
                End If
            Else
                Using dlg As New OpenFileDialog()
                    dlg.Filter = "XML Files|*.xml"
                    If dlg.ShowDialog() = DialogResult.OK Then
                        result = True
                        View.XmlFilePath = dlg.FileName
                    End If
                End Using
            End If
            Return result
        End Function

        Public Overrides Sub Commit()
            Model.Tag = View.XmlFilePath
            Model.DataSchema = New Person()
        End Sub
        Public Overrides ReadOnly Property MoveNextEnabled() As Boolean
            Get
                Return False
            End Get
        End Property
        Public Overrides ReadOnly Property FinishEnabled() As Boolean
            Get
                Return True
            End Get
        End Property
    End Class
End Namespace
